Fix shellcheck errors in scripts (part 2)

MuteSpirit opened the pull request:
Fixing shellcheck issues in 2nd part of scripts mentioned in task #915

Description of fixed shellcheck issues you may found in commit comments

Commit:
scripts: docker-runner.sh: fix shellcheck issues

shellcheck scan of docker-runner.sh revealed few issues:
Line 3: SC2164 - no check if cd successful
Line 4: SC2086 - missing quotes around scriptdir variable

To resolve:
Line 3: add exit if cd fails
Line 4: add quotes around whole run.sh call

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: docker-builder.sh: fix shellcheck issues

During shellcheck scan of docker-builder.sh script, few issues
raised:
Line 3: SC2164 - no check if cd fails
Line 4: SC2086 - missing quotes around "scriptdir" variable

In oder to fix:
Line 3: exit if cd fails
Line 4: Add quotes around source command

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: ci/owncloud/upload_to_owncloud.sh: fix shellcheck issues

shellcheck scan of upload_to_owncloud.sh script revealed missing double
quotes around "local_path" variables.

Add double quotes around "local_path" variable.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: ci/run_easymesh_cert.sh: fix shellcheck issues

shellcheck scan revealed missing double quotes for $0 variable.

Add double quotes for $0 variable.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: tests/openwrt/test_status.sh: fix shellcheck issues

shellcheck scan of test_status.sh found missing quotes for $0 variable.

Add quotes for $0 variable.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: tools/docker/builder/openwrt/build.sh: fix shellcheck issues

schellcheck scan of openwrt build.sh script revealed next issues:
Line 6: SC1090 - shellcheck unable to follow sourced file
Line 9: SC2086 - missing quotes for $0 variable
Line 40: SC2086 - missing quotes for BUILD_OPTIONS variable
Line 74: SC2006, SC2181 - usage of legacy backticks and indirect return
code check of "getopt"

In order to fix:
Line 6: add shellcheck directive to point shellcheck to sourced file
Line 9: add double quotes around $0 variable
Line 40: add quotes around "BUILD_OPTIONS" variable and pass it as
single line
Line 74: place getop call in if condition to check return code directly

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: tools/docker/build.sh: fix shellcheck issues

shellcheck scan revealed bunch of issues in tools/docker/build.sh
script:
Line 9: SC2164 - no check if cd succeeded
Line 12: SC1090, SC2086 - shellcheck unable to follow sourced file and
missing double quotes
Line 17: missing quotes around "scriptdir" variable
Line 28: "! -z" instead of "-n" in test statement
Line 34: SC2086 - missing quotes around "docker_args" and "$@"

In order to fix issues:
Change shell from sh to bash, for array functionality.
Line 9: add exit if cd failed
Line 12: add shellcheck directive to source file
put source command in double quotes
Line 17: add quotes for "scriptdir" variable
Refactor docker args to improve readiness
Line 28: replace "! -z" with direct "-n" check
Line 34: pass "docker_args" as array and add quotes around "$@"

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: tools/docker/image-build.sh: remove "--native" option support

Option "--native" of "tools/docker/image-build.sh" makes
to use it the same base OS version image as the running system is.
Looks, like it's not used anymore.

Remove code related to that option.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: tools/docker/image-build.sh: fix shellcheck issues

shellcheck scan tools/docker/image-build.sh revealed few issues:
Line 9: SC2164 -  no check if cd is successful
Line 12: SC1090, SC2086 - shellcheck unable to follow sourced file
missing quotes around "scriptdir" variable
Line 15: SC2086 - missing quotes around "$0"
Line 27: SC2006, SC2182 - usage of backticks instead of $(..)
indirect check of getopt return code
Line 49,51,56,62: SC2086 - missing quotes around $IMAGE variable
Line 50,57,63: SC2086 - missing quotes around $TAG variable
Line 51,64: SC2086 - missing quotes around $scriptdir
Line 71: SC2068 - no double quotes around $@

In order to fix:
Line 9: exit if cd fails
Line 12: add shellcheck directive to follow sourced file
Line 15: add quotes for $0
Line 27: put getopt call in if condition and replace backticks with
$(..)
Line 49,51,56,62: add quotes around $IMAGE
Line 50,57,63: add quotes around $TAG
Line 51,64: add quotes around $scriptdir
Line 71: add quotes around "$@"

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

rmelotte commented on tools/docker/builder/openwrt/build.sh:
```diff
@@ -82,7 +85,7 @@ main() {
         case "$1" in
             -h | --help)               usage; exit 0; shift ;;
-            -b | --build-options)      BUILD_OPTIONS="$2"; shift; shift ;;
+            -b | --build-options)      BUILD_OPTIONS+=($(echo -n "$2")); shift; shift ;;
```
Here I think the safest is to drop '--build-options' entirely. I think nobody use it anymore.

```suggestion
```

MuteSpirit replied:
Ok, i'll do that if almost one more reviewer will approve your suggestion

arnout replied:
I agree!

abelog replied:
Than drop it is. This option is removed.

rmelotte commented on docker-builder.sh:
```diff
@@ -1,4 +1,4 @@
 
-scriptdir="$(cd "${0%/*}"; pwd)"
-${scriptdir}/tools/docker/build.sh "$@"
+scriptdir="$(cd "${0%/*}" || exit 1; pwd)"
```
Like for the other PR, I think `$(cd "${0%/*}" && pwd)` is less confusing (exit might make you think it exits the script)
(nitpick)

rmelotte commented on tools/docker/builder/openwrt/build.sh:
```diff
@@ -1,12 +1,13 @@
 scriptdir="$(cd "${0%/*}"; pwd)"
 rootdir="${scriptdir%/*/*/*/*}"
 
+# shellcheck source=../../../../tools/functions.sh
```
```suggestion
# shellcheck source=../../../functions.sh
```

arnout replied:
I disagree - with `../../../../tools/functions.sh` it corresponds one-to-one with the expansion of `$rootdir/tools/functions.sh`

rmelotte commented on tools/docker/image-build.sh:
```diff
@@ -6,27 +6,29 @@
 rootdir="${scriptdir%/*/*}"
 
-. ${rootdir}/tools/functions.sh
+# shellcheck source=../../tools/functions.sh
```
```suggestion
# shellcheck source=../functions.sh
```

arnout replied:
I disagree

rmelotte reviewed (approved):
Only nitpicks, LGTM :+1:

arnout commented on docker-runner.sh:
```diff
@@ -1,4 +1,4 @@
 
-scriptdir="$(cd "${0%/*}"; pwd)"
-${scriptdir}/tools/docker/run.sh "$@"
+scriptdir="$(cd "${0%/*}" || exit 1; pwd)"
```
As mentioned elsewhere: I prefer
```suggestion
scriptdir="$(cd "${0%/*}" && pwd)"
```
but that's really personal preference.

arnout commented on tools/docker/builder/openwrt/build.sh:
```diff
@@ -1,12 +1,13 @@
-#!/bin/sh -e
+#!/bin/bash -e
```
Please mention something like this very explicitly in the commit message. Not just that we changed the shell to bash, but also why.

arnout commented on tools/docker/builder/openwrt/build.sh:
```diff
@@ -72,7 +72,7 @@ main() {
     fi
 
-    if ! OPTS=$(getopt -o 'hb:d:io:r:t:v' --long help,build-options:,device:,image,openwrt-version:,openwrt-repository:,tag:,verbose -n 'parse-options' -- "$@"); then
+    if ! OPTS=$(getopt -o 'hb:d:io:r:t:v' --long help,build-options:,target-device:,image,openwrt-version:,openwrt-repository:,tag:,verbose -n 'parse-options' -- "$@"); then
```
Nice catch!

arnout commented on tools/docker/build.sh:
```diff
@@ -1,37 +1,38 @@
-#!/bin/sh
+#!/bin/bash
```
Again, please explain why in commit message.

arnout commented on tools/docker/build.sh:
```diff
@@ -1,37 +1,38 @@
 ###############################################################
 
-scriptdir="$(cd "${0%/*}"; pwd)"
+scriptdir="$(cd "${0%/*}" || exit 1; pwd)"
```
Again, slight preference for `&&`

arnout commented on tools/docker/build.sh:
```diff
@@ -1,37 +1,38 @@
-    --entrypoint=./tools/maptools.py \
-    "
+    docker_args=(
+        --workdir "${rootdir}"
+        --user "${SUDO_UID:-$(id -u)}:${SUDO_GID:-$(id -g)}"
+        -e "USER=${SUDO_USER:-${USER}}"
+        -v "${rootdir}:${rootdir}"
+        --entrypoint "./tools/maptools.py"
```
I really like how it's much easier to read now!

arnout commented on tools/docker/build.sh:
```diff
@@ -1,37 +1,38 @@
-    -e USER=${SUDO_USER:-${USER}} -v ${rootdir}:${rootdir} \
-    --entrypoint=./tools/maptools.py \
-    "
+    docker_args=(
```
(nitpick) Now we're using bash, we can also 
```suggestion
    local docker_args=(
```

arnout commented on tools/docker/image-build.sh:
```diff
@@ -19,12 +19,11 @@ usage() {
     echo "      -h|--help - show this help menu"
     echo "      -v|--verbose - verbosity on"
     echo "      -b|--base-image - Base OS image to use (Dockerfile 'FROM')"
-    echo "      -n|--native - Use the same base OS image as the running system"
```
As mentioned in the other PR: Please include an explanation in the commit message why this didn't work.

arnout commented on tools/docker/image-build.sh:
```diff
@@ -6,13 +6,14 @@
 ###############################################################
 
-scriptdir="$(cd "${0%/*}"; pwd)"
+scriptdir="$(cd "${0%/*}" || exit 1; pwd)"
```
```suggestion
scriptdir="$(cd "${0%/*}" && pwd)"
```

arnout reviewed (approved):
Many nitpicks, nothing major.

abelog committed:
.gitlab-ci.yaml: remove -v option from .build-for-openwrt

GitLab CI is failing due to usage of -v option of tools/docker/builder/openwrt/build.sh
script. Recent changes have removed this option.

Adjust .gitlab-ci.yaml to recent changes by stripping -v option from
call of build.sh in .build-for-openwrt stage.

Signed-off-by: Anton Bilohai <a.bilohai@inango-systems.com>
